### PR TITLE
📈 The start of token tracking

### PR DESCRIPTION
**Pull request by awtkns**

### CODE REVIEW

File: platform/reworkd_platform/services/tokenizer/dependencies.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from fastapi import Request
+
+from reworkd_platform.services.tokenizer.service import TokenService
```
Comment (asim-shrestha) [inline platform/reworkd_platform/services/tokenizer/dependencies.py:3]: TokenizerService?

File: platform/reworkd_platform/services/tokenizer/lifetime.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import tiktoken
+from fastapi import FastAPI
+
+ENCODING_NAME = "cl100k_base"  # gpt-4, gpt-3.5-turbo, text-embedding-ada-002
```
Comment (asim-shrestha) [inline platform/reworkd_platform/services/tokenizer/lifetime.py:4]: Will need some mechanism to change this based on ModelSettings in the future.

File: platform/reworkd_platform/services/tokenizer/service.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        return self.encoding.encode(text)
+
+    def detokenize(self, tokens: list[int]) -> str:
+        return self.encoding.decode(tokens)
+
+    def token_count(self, text: str) -> int:
+        return len(self.tokenize(text))
+
+    def get_context_space(
+        self, prompt: str, max_tokens: int, reserved_response_tokens: int
```
Comment (asim-shrestha) [inline platform/reworkd_platform/services/tokenizer/service.py:18]: The model object itself may have a max_tokens defined, not sure

**Review by asim-shrestha — APPROVED** [review]

File: platform/reworkd_platform/services/tokenizer/lifetime.py (hunk 2 above)
Comment (awtkns) [reply platform/reworkd_platform/services/tokenizer/lifetime.py:4]: I think we can look at this in the future, right now tiktoken only seems to support openai

**Pull request merged**